src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -58,56 +58,86 @@ async function kdf(hash_algo, X, length, param) {
   return digest.subarray(0, length);
 }
 
-
 /**
- * Encrypt and wrap a session key
+ * Generate ECDHE ephemeral key and secret from public key
  *
  * @param  {module:type/oid}        oid          Elliptic curve object identifier
  * @param  {module:enums.symmetric} cipher_algo  Symmetric cipher to use
  * @param  {module:enums.hash}      hash_algo    Hash algorithm to use
- * @param  {module:type/mpi}        m            Value derived from session key (RFC 6637)
  * @param  {Uint8Array}             Q            Recipient public key
  * @param  {String}                 fingerprint  Recipient fingerprint
- * @returns {Promise<{V: BN, C: BN}>}            Returns ephemeral key and encoded session key
+ * @returns {Promise<{V: BN, Z: Uint8Array}>}    Returns public part of ephemeral key and generated ephemeral secret
  * @async
  */
-async function encrypt(oid, cipher_algo, hash_algo, m, Q, fingerprint) {
+async function genPublicEphemeralKey(oid, cipher_algo, hash_algo, Q, fingerprint) {
   const curve = new Curve(oid);
   const param = buildEcdhParam(enums.publicKey.ecdh, oid, cipher_algo, hash_algo, fingerprint);
   cipher_algo = enums.read(enums.symmetric, cipher_algo);
   const v = await curve.genKeyPair();
   Q = curve.keyFromPublic(Q);
   const S = v.derive(Q);
-  const Z = await kdf(hash_algo, S, cipher[cipher_algo].keySize, param);
-  const C = aes_kw.wrap(Z, m.toString());
   return {
     V: new BN(v.getPublic()),
-    C: C
+    Z: await kdf(hash_algo, S, cipher[cipher_algo].keySize, param)
```

Comment:
These are becoming increasingly nitpickier nitpicks, but if you add `const V = new BN(v.getPublic());` above and restore the `const Z = ...` as well, you can `return { V, Z };` here and the linter shouldn't complain.